detox/src/devices/android/ADB.js

```diff
@@ -287,6 +269,59 @@ class ADB {
   }
 }
 
+class AdbDevicesParser {
```

Comment:
Why not adhere to the built-in `AsyncIterator` and [for await..of](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of)? On one hand, `for await of` is supported only by Node 10.x, but if we implement AdbParser as AsyncIterable right now, then we can easily rewrite your findDevice() and devices() the moment we jump onto node>=10.x